slp2mp4/gui_test.py

```diff
@@ -0,0 +1,289 @@
+    selected_widescreen.set(widescreen_option.get())
+
+def update_combine_option(*args):
+    selected_combine.set(combine_option.get())
+
+def update_remove_slps_option(*args):
+    selected_remove_slps.set(remove_slps_option.get())
+
+def update_remove_short_slps_option(*args):
+    selected_remove_short_slps.set(remove_short_slps_option.get())
+
+def browse_video_output():
+    # Opens a directory selection window
+    video_output_path = filedialog.askdirectory(title="Select Video Output Folder")
+
+    # Update the video output variable with the selected path
+    video_output_var.set(video_output_path)
+
+# Create the main window
+root = tk.Tk()
+root.title("SLP2MP4 - Iowa Melee")
+
+# Variables to store user entries
+melee_iso_var = tk.StringVar()
+dolphin_directory_var = tk.StringVar()
+slippi_folder_var = tk.StringVar()
+selected_encoder = tk.StringVar()
+selected_resolution = tk.StringVar(value="1080p")  # Default value is set to 1080p
+selected_widescreen = tk.StringVar(value="False")  # Default value is set to False
+selected_combine = tk.StringVar()
+selected_remove_slps = tk.StringVar(value="False")  # Default value is set to False
+selected_remove_short_slps = tk.StringVar(value="False")  # Default value is set to False
+bitrate_var = tk.StringVar(value="20000")  # Default value is set to 20000
+
+#script and data folder path variables
+script_directory = os.path.dirname(os.path.realpath(__file__))
+data_folder = os.path.join(script_directory, "data")
+
+config_data = {
+    "melee_iso": melee_iso_var.get(),
+    "dolphin_dir": dolphin_directory_var.get(),
+    "resolution": selected_resolution.get(),
+    "video_backend": selected_encoder.get(),
+    "widescreen": selected_widescreen.get(),
+    "bitrateKbps": bitrate_var.get(),
+    "parallel_games": "recommended",
+    "remove_short": selected_remove_short_slps.get(),
+    "combine": selected_combine.get(),
+    "remove_slps": selected_remove_slps.get(),
+    "ffmpeg": "C:\\ffmpeg\\bin\\ffmpeg.exe"
+    # Can add other configuration parameters here as needed
+}
+
+
+
+def save_config():
+    config_data["melee_iso"] = melee_iso_var.get()
+    config_data["dolphin_dir"] = dolphin_directory_var.get()
+    config_data["resolution"] = selected_resolution.get()
+    config_data["video_backend"] = selected_encoder.get()
+    config_data["widescreen"] = selected_widescreen.get()
+    config_data["bitrateKbps"] = bitrate_var.get()
+    config_data["remove_short"] = selected_remove_short_slps.get()
+    config_data["combine"] = selected_combine.get()
+    config_data["remove_slps"] = selected_remove_slps.get()
+    # Add other config parameters here if needed
+
+    os.makedirs(data_folder, exist_ok=True)  # Ensure the "data" folder exists
+    config_file_path = Config(False).paths.config_json
+
+    with open(config_file_path, "w", encoding="utf-8") as config_file:
+        json.dump(config_data, config_file, indent=4)
+        config_file.truncate()
```

Comment:
This function could be moved to `config.py`, modified to accept a dict, then it could be used in `slp2mp4.py` as well